src/overlays/actors/ovl_Bg_Hakugin_Post/z_bg_hakugin_post.c

```diff
@@ -21,103 +54,994 @@ const ActorInit Bg_Hakugin_Post_InitVars = {
+            if (Math3D_XZLengthSquared(unkStruct->unk_0000[i].unk_14.x, unkStruct->unk_0000[i].unk_14.z) > 278784.03f) {
+                func_80A9B554(this, globalCtx, unkStruct, &unkStruct->unk_0000[i]);
+                func_8019F128(NA_SE_EV_GLASSBROKEN_IMPACT);
+                unkStruct->unk_0000[i].unk_34 = 4;
+                unkStruct->unk_0000[i].unk_30 = 30;
+            }
+        } else if (unkStruct->unk_0000[i].unk_34 == 4) {
+            unkStruct->unk_0000[i].unk_30--;
+            if (unkStruct->unk_0000[i].unk_30 <= 0) {
+                unkStruct->unk_0000[i].unk_34 = 5;
+            }
+        }
+    }
+}
+
+void func_80A9C058(BgHakuginPost* this, GlobalContext* globalCtx, BgHakuginPostUnkStruct* unkStruct) {
+    s32 i;
+    s32 pad;
+    Vec3f sp44;
+    s32 pad2[2];
+    s16 quake;
+
+    if (this->unk_170 == 0) {
+        for (i = 0; i < unkStruct->count; i++) {
+            BgHakuginPostUnkStruct1* unkStruct1 = &unkStruct->unk_0000[i];
+
+            if (unkStruct1->unk_2F & 2) {
+                sp44.x = this->dyna.actor.home.pos.x + unkStruct1->unk_14.x;
+                sp44.y = this->unk_16C + unkStruct1->unk_14.y;
+                sp44.z = this->dyna.actor.home.pos.z + unkStruct1->unk_14.z;
+                func_8013ECE0(Math3D_DistanceSquared(&sp44, &PLAYER->actor.world.pos), 255, 20, 150);
+                quake = Quake_Add(ACTIVE_CAM, 3);
+                Quake_SetSpeed(quake, 20000);
+                Quake_SetQuakeValues(quake, 7, 0, 0, 0);
+                Quake_SetCountdown(quake, 12);
+                if (this->unk_179 <= 0) {
+                    func_8019F128(NA_SE_EV_STONEDOOR_STOP);
+                    this->unk_179 = 40;
+                }
+                break;
+            }
+        }
+    }
+}
+
+void func_80A9C18C(BgHakuginPost* this, GlobalContext* globalCtx) {
+    s32 pad;
+    Player* player = PLAYER;
+    Camera* activeCam = ACTIVE_CAM;
+    s16 quake;
+
+    func_8013ECE0(Math3D_XZDistanceSquared(player->actor.world.pos.x, player->actor.world.pos.z,
+                                           this->dyna.actor.home.pos.x, this->dyna.actor.home.pos.z),
+                  255, 20, 150);
+    quake = Quake_Add(activeCam, 3);
+    Quake_SetSpeed(quake, 17232);
+    Quake_SetQuakeValues(quake, 6, 0, 0, 0);
+    Quake_SetCountdown(quake, 20);
+}
+
+void func_80A9C228(BgHakuginPost* this, GlobalContext* globalCtx, BgHakuginPostUnkStruct* unkStruct) {
+    s32 val;
+    BgHakuginPostUnkStruct1* spC8 = NULL;
+    BgHakuginPostUnkStruct1* spC4;
+    Vec3f spB8;
+    Vec3f spAC;
+    Vec3f spA0;
+    s32 i;
+    f32 temp_f20;
+    s16 temp_s0;
+    s32 j;
+
+    for (i = 0; i < unkStruct->count; i++) {
+        spC4 = spC8;
+        spC8 = &unkStruct->unk_0000[i];
+
+        if (unkStruct->unk_0000[i].unk_34 == 2) {
+            for (j = 0, val = 0; j < 3; j++, val += 0x5555) {
+                temp_f20 = Rand_ZeroOne() * 264.0f;
+                temp_s0 = (Rand_Next() & 0xFFF) + val;
```

Comment:
I wonder if this would match too
```suggestion
                temp_s0 = (Rand_Next() % 0x1000) + val;
```

Reply:
Doesn't match due to being signed.